components/speedreader/rust/lib/src/readability/src/extractor.rs

```diff
@@ -29,12 +36,33 @@ where
     extract_dom(&mut dom, url, &HashMap::new())
 }
 
+pub fn preprocess<R>(input: &mut R) -> Result<Product, std::io::Error>
+where
+    R: Read,
+{
+    let mut dom = parse_document(RcDom::default(), Default::default())
+        .from_utf8()
+        .read_from(input)?;
+
+    let mut title = Title::default();
+    let handle = dom.document.clone();
+    scorer::preprocess(&mut dom, handle, &mut title);
+    let mut bytes = vec![];
+    let document: SerializableHandle = dom.document.clone().into();
+    serialize(&mut bytes, &document, serialize::SerializeOpts::default())?;
+    let content = String::from_utf8(bytes).unwrap_or_default();
+    Ok(Product {
+        title: title.title,
+        content: content,
```

Comment:
Since you're initializing the `content` field using a variable called `content`, you can actually just use the shorthand form, i.e.:
```rust
Product {
    title: title.title,
    content,
}
```